### PR TITLE
Proxy and User-Agent helpers.

### PR DESCRIPTION
**Pull request by adamcik**

### CODE REVIEW

**Comment by jodal** [comment]
Looks good to me, but I want us to agree on a plan for how to separate public and private utils, both in import path and docs, before merging this.

**Comment by adamcik** [comment]
Agreed, this was my caveat as well. But as a quick experiment I went ahead yesterday and converted Dirble and Spotify to give it a go. Which showed my we need the auth option and that I had a bug concerning the scheme.

**Comment by adamcik** [comment]
Other question is if we should check for `user` and `pass` or just `user`.

**Comment by jodal** [comment]
Given #1173, I think this should be moved to e.g. `mopidy.httpclient` and included in the new "Utililties" section on the API docs page.

**Comment by adamcik** [comment]
Done.

File: tests/test_httpclient.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from __future__ import unicode_literals
+
+import re
+
+import pytest
+
+from mopidy.utils import httpclient
```
Comment (jodal) [inline tests/test_httpclient.py]: The tests will fail on this import.

**Comment by adamcik** [comment]
And fixed, didn't notice the output from my `py.test -f`...

**Pull request merged**